File: 6 Reverse Integer/Solution.java
class Solution {
    public int reverse(int x) {
        int[] arrX = convertTOArray(x);
        int iX = 0;
        for (int i = arrX.length-1; i>=0; i--){
            iX = iX * 10 + arrX[i]; 
        } 
        return iX;
    }
    private int[] convertTOArray(int num){
            int numDigit = (int) Math.log10(num) + 1;
            int[] arrNum = new int[numDigit];
            for(int i =0; i < numDigit; i++){
                arrNum[i] = num % 10;
                num = num /10;
            }
        return arrNum;
    }
}

File: 6 Reverse Integer/Solution.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x <0 else 1
        x = abs(x)

        reversed_x = 0

        while x != 0:
            digit = x % 10

            if reversed_x >(2**31 -1 - digit) // 10:
                return 0
        
            reversed_x = reversed_x * 10 + digit
            x = x // 10

        return sign * reversed_x